processing/src/main/java/org/apache/druid/segment/join/JoinableFactoryWrapper.java

```diff
@@ -144,9 +144,15 @@ static JoinClauseToFilterConversion convertJoinToFilter(
       final Set<String> rightPrefixes
   )
   {
+    // This optimization kicks in when there is exactly 1 equijoin
+    // The reason being that getNonNullColumnValues uses a TreeSet for handling duplicates
+    // and does not return values in order.
+    // In case TreeSet if replaced by a LinkedHashSet, duplicates are not handled correctly
```

Comment:
This comment doesn't really make sense to someone that hasn't seen previous versions of this PR. Consider removing or rewording it.

Reply:
Addressed